crates/task-impls/src/consensus/mod.rs

```diff
@@ -498,6 +501,12 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusTaskState<TYPES, I>
                 if disperse.data.recipient_key != self.public_key {
                     return;
                 }
+                let Some(proposal) = self.current_proposal.clone() else {
+                    return;
+                };
+                if proposal.get_view_number() != view {
+                    return;
+                }
```

Comment:
Does this actually do anything? Since we now check at the start of `spawn_vote_task` anyway.

Reply:
No but I don't mind being a bit redundant to reduce the odds of messing this up

Reply:
I'm happy to change it if you think we should

Reply:
We don't check the view number though

Reply:
I added the view check in spawn_vote as well

Reply:
Oh, duh, I misread "now", sorry.